### ISSUE TITLE
Is there a way or an option that I can compute similarity score after calling inference to calculate embeddings?

### ISSUE THREAD

**Issue by zwsjink** [rom1504/clip-retrieval#305]
Well, I use
'''
clip-retrieval inference --input_dataset image/mytest.tar	--output_folder embeddings_folder  												--clip_model ViT-L/14  --input_format="webdataset"
'''
to successfully calculate embeddings for both images and texts, and I do see npy files generated. However, I wish there is a way that the similarity score can be calculated at the same time and store to the generated parquet file in the meantime.

**Comment by rom1504** [comment]
You can use embedding-reader to compute such similarities between pairs
Computing these dot products is very cheap

I guess we could also add the option to do it here too though

**Comment by zwsjink** [comment]
> embedding-reader

I see, that would be great. By the way, in the paper "_DataComp: In search of the next generation of multimodal datasets_"  I see you guys are using cosine similarity,  how big difference is it between cosine similarity and  dot product (maybe after normalization?).

**Comment by rom1504** [comment]
After normalization dot product is cosine

**Comment by nicolas-dufour** [comment]
Hey, @rom1504 , is there an option to store the clip scores when using doing the clip inference transformation? Thanks!

**Comment by rom1504** [comment]
I guess you mean to compute dot products between pair of text and image
embeddings ?
There is no such option in clip-retrieval inference. But it's very cheap to
compute those on cpu after the inference is done so maybe just do that ?

**Comment by nicolas-dufour** [comment]
Yeah but i would like to avoid keeping the image embeddings to save storage.
